src/cbc_sdk/workload/vulnerability_assessment.py

```diff
@@ -151,37 +52,26 @@ def _refresh(self):
         Returns:
             bool: True if refresh was successful, False if not.
         """
-        resp = self._cb.get_object(self._build_api_request_uri())
-        resp['cve_id'] = self._model_unique_id
-        self._info = resp
-        self._last_refresh_time = time.time()
-        return True
-
-    def _build_api_request_uri(self, http_method="GET"):
-        """
-        Build the unique URL used to make requests for this object.
+        result = list(self._cb.select(Vulnerability).add_criteria("cve_id", self._model_unique_id))
```

Comment:
I might not be reading this correctly, but to get a specific vulnerability we are using the search and not the : https://developer.carbonblack.com/reference/carbon-black-cloud/workload-protection/latest/vulnerability-assessment/#get-vulnerability-details in fact I think this API (Get Vulnerability Details) is not used at all.

Reply:
If you look at the responses between the search and the vulnerability details, the search response contains the vulnerability details however it is `null` when the following is not included in the response `/_search?dataForExport=true`. So we can get both in a single call instead of needing to make 1 + n api calls when fetching the Vulnerabilities. The other aspect I noticed was the vulnerability details does not contain the necessary properties in order to make additional helper calls like get affected_assets as you need the os_product_id. 

The end result was finding that the _search request supported the cve_id property so we could get the exact cve vulnerability while keep the number of API calls and providing the entire vulnerability object